producers/src/main/java/uk/gov/moj/sdt/producers/sdtws/SdtEndpointPortType.java

```diff
@@ -1,263 +1,270 @@
+                    detail.append("\n\terror code=" +
+                            response.getBulkRequestStatus().getBulkStatus().getError().getCode() +
+                            "\n\terror description=" +
+                            response.getBulkRequestStatus().getBulkStatus().getError().getDescription() + "\n");
+                }
+            }
+
+            // Write message to 'performance.log' for this logging point.
+            PerformanceLogger.log(this.getClass(), PerformanceLogger.LOGGING_POINT_9,
+                    "Bulk feedback response returned", detail.toString());
+        }
+
+        return response;
+    }
+
+    @Override
+    public SubmitQueryResponseType submitQuery(final SubmitQueryRequestType submitQueryRequest) {
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Endpoint called for submit query by customer [" +
+                    submitQueryRequest.getHeader().getSdtCustomerId() + "]");
+        }
+
+        if (PerformanceLogger.isPerformanceEnabled(PerformanceLogger.LOGGING_POINT_2)) {
+            final StringBuffer detail = new StringBuffer();
+            detail.append("\n\n\tcustomer=" + submitQueryRequest.getHeader().getSdtCustomerId() +
+                    "\n\ttarget application=" + submitQueryRequest.getHeader().getTargetApplicationId() + "\n");
+
+            // Write message to 'performance.log' for this logging point.
+            PerformanceLogger.log(this.getClass(), PerformanceLogger.LOGGING_POINT_2,
+                    "Submit query request received", detail.toString());
+        }
+
+        SubmitQueryResponseType response = null;
+
+        try {
+            response = wsReadSubmitQueryHandler.submitQuery(submitQueryRequest);
+        }
+        catch (Exception throwable)
+        {
+            handleThrowable(throwable);
+        }
+
+        if (PerformanceLogger.isPerformanceEnabled(PerformanceLogger.LOGGING_POINT_9)) {
+            final StringBuffer detail = new StringBuffer();
+            detail.append("\n\n\tbulk request status=" + response.getSdtService() + "\n\tsdt customer id=" +
+                    response.getSdtCustomerId() + "\n");
+            if (response.getStatus().getError() != null) {
+                detail.append("\n\terror code=" + response.getStatus().getError().getCode() +
+                        "\n\terror description=" + response.getStatus().getError().getDescription() + "\n");
+            }
+
+            // Write message to 'performance.log' for this logging point.
+            PerformanceLogger.log(this.getClass(), PerformanceLogger.LOGGING_POINT_9,
+                    "Submit query response returned", detail.toString());
+        }
+
+        return response;
+
+    }
+
+    /**
+     * Handles throwable and re-throws runtime exception.
+     *
+     * @param throwable exception to be handled
+     */
+    private void handleThrowable(final Throwable throwable) {
```

Comment:
Should this be renamed as handleException() now that it's being called with an Exception instead of Throwable?  If so then don't forget to change parameter type and name and JavaDoc comment.